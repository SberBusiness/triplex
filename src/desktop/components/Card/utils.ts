import {ECardRoundingSize} from '@sberbusiness/triplex/desktop/components/Card/enums';

/** Соответствие типа скругления карточки стилевому классу. */
export const mapCardRoundingSizeToCssClass = {
    [ECardRoundingSize.MD]: 'cssClass[roundingMD]',
    [ECardRoundingSize.SM]: 'cssClass[roundingSM]',
};
