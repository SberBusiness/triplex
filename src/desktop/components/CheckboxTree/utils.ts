import {ICheckboxTreeCheckboxData} from './types';

/** Обход ICheckboxData[]. */
export const traverseCheckboxes = (checkboxes: ICheckboxTreeCheckboxData[], cb: (checkbox: ICheckboxTreeCheckboxData) => void): void => {
    checkboxes.forEach((c) => {
        if (c.children) traverseCheckboxes(c.children, cb);
        cb(c);
    });
};

/** Обновление флага checked и bulk родителя, при изменении дочернего чекбокса. */
export const checkParentCheckboxes = (checkbox: ICheckboxTreeCheckboxData): void => {
    if (!checkbox.children) {
        return;
    }

    let checkedChildrenCount = 0;
    let bulkChildrenCount = 0;

    checkbox.children.forEach((c) => {
        c.checked ? checkedChildrenCount++ : '';
        c.bulk ? bulkChildrenCount++ : '';
    });

    // Все дочерние чекбоксы выбраны.
    if (checkedChildrenCount === checkbox.children.length) {
        checkbox.checked = true;
        checkbox.bulk = bulkChildrenCount !== 0;
    } else if (checkedChildrenCount > 0) {
        // Некоторые дочерние чекбоксы выбраны.
        checkbox.checked = true;
        checkbox.bulk = true;
    } else {
        // Все дочерние чекбоксы не выбраны.
        checkbox.checked = false;
    }
};

/** Обновление флага checked дочерних чекбоксов, при изменении родителя. */
export const checkChildrenCheckboxes = (checkbox: ICheckboxTreeCheckboxData): void => {
    if (!checkbox.children) {
        return;
    }

    checkbox.children.forEach((c) => {
        c.checked = checkbox.checked;
        checkChildrenCheckboxes(c);
    });
};
