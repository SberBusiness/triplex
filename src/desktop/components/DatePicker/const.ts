/** Формат даты, отображаемый при вооде. */
export const inputDateFormat = 'DD.MM.YYYY';
