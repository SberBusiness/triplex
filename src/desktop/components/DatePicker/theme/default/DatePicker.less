@import '../../../../common/theme/default';

.datePicker {
    position: relative;
    width: @picker-width;
}
