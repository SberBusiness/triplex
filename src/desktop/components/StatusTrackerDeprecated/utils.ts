import {
    EStatusTrackerDeprecatedStepStatus,
    StatusTrackerDeprecatedStatus,
    StatusTrackerDeprecatedStepNumber,
} from '@sberbusiness/triplex/desktop/components/StatusTrackerDeprecated/enums';
import {EStepStatus} from '@sberbusiness/triplex/desktop/components/Step/enums';

/** Мап общего статуса компонента в статус конкретного шага.  */
const trackerStatusToStepStatusMap = {
    [StatusTrackerDeprecatedStatus.WAIT]: EStatusTrackerDeprecatedStepStatus.WAIT,
    [StatusTrackerDeprecatedStatus.WARNING]: EStatusTrackerDeprecatedStepStatus.WARNING,
    [StatusTrackerDeprecatedStatus.SUCCESS]: EStatusTrackerDeprecatedStepStatus.SUCCESS,
    [StatusTrackerDeprecatedStatus.ERROR]: EStatusTrackerDeprecatedStepStatus.ERROR,
};

/** Маппинг общего статуса компонента в статус конкретного шага.  */
export const mapTrackerStatusToStepStatus = (
    currentStep: StatusTrackerDeprecatedStepNumber,
    status: StatusTrackerDeprecatedStatus,
    iteratorStep: StatusTrackerDeprecatedStepNumber
): EStatusTrackerDeprecatedStepStatus => {
    let result;

    if (currentStep > iteratorStep) {
        result = EStatusTrackerDeprecatedStepStatus.SUCCESS;
    } else if (currentStep < iteratorStep) {
        result = EStatusTrackerDeprecatedStepStatus.DISABLED;
    } else {
        result = trackerStatusToStepStatusMap[status];
        if (result === undefined) {
            throw Error(`StatusTrackerStatus not mapped: '${status}'.`);
        }
    }

    return result;
};

/**
 * Мапа статусов шагов трекера в шаги маркера.
 *
 * TODO: Выпилить в ближайшем мажорном релизе, перевести на общий энам.
 */
const statusTrackerStepStatusToStepMarkerStatusMap = {
    [EStatusTrackerDeprecatedStepStatus.WAIT]: EStepStatus.WAIT,
    [EStatusTrackerDeprecatedStepStatus.WARNING]: EStepStatus.WARNING,
    [EStatusTrackerDeprecatedStepStatus.ERROR]: EStepStatus.ERROR,
    [EStatusTrackerDeprecatedStepStatus.SUCCESS]: EStepStatus.SUCCESS,
    [EStatusTrackerDeprecatedStepStatus.DISABLED]: EStepStatus.DISABLED,
};

/**
 * Мапинг статусов шагов трекера в шаги маркера.
 *
 * TODO: Выпилить в ближайшем мажорном релизе, перевести на общий энам.
 */
export const mapStatusTrackerStepStatusToStepMarkerStatus = (status: EStatusTrackerDeprecatedStepStatus): EStepStatus => {
    const result = statusTrackerStepStatusToStepMarkerStatusMap[status];

    if (result === undefined) {
        throw Error(`EStatusTrackerStepStatus not mapped: '${status}'.`);
    }

    return result;
};
