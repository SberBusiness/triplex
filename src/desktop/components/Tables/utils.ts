import {EScreenWidth} from '@sberbusiness/triplex/common/enums/EScreenWidth';
import {ECellType, EHorizontalAlign, EVerticalAlign} from '@sberbusiness/triplex/desktop/components/Tables/TableBasic/enums';
import {cssClass} from '@sberbusiness/triplex/desktop/utils/cssClass';

/**
 * Преобразование значения горизонтального выравнивания в css-класс.
 * @param {EHorizontalAlign} [horizontalAlign] Значение для маппинга.
 */
export const mapHorizontalAlignToClassName = (horizontalAlign: EHorizontalAlign = EHorizontalAlign.LEFT): string => {
    switch (horizontalAlign) {
        case EHorizontalAlign.LEFT:
            return 'cssClass[alignLeft]';
        case EHorizontalAlign.RIGHT:
            return 'cssClass[alignRight]';
        case EHorizontalAlign.CENTER:
            return 'cssClass[alignCenter]';
    }
};

/**
 * Преобразование значения вертикального выравнивания в css-класс.
 * @param {EVerticalAlign} [verticalAlign] Значение для маппинга.
 */
export const mapVerticalAlignToClassName = (verticalAlign: EVerticalAlign = EVerticalAlign.BASELINE): string => {
    switch (verticalAlign) {
        case EVerticalAlign.BASELINE:
            return 'cssClass[verticalAlignBaseline]';
        case EVerticalAlign.SUB:
            return 'cssClass[verticalAlignSub]';
        case EVerticalAlign.SUPER:
            return 'cssClass[verticalAlignSuper]';
        case EVerticalAlign.TEXT_TOP:
            return 'cssClass[verticalAlignTextTop]';
        case EVerticalAlign.TEXT_BOTTOM:
            return 'cssClass[verticalAlignTextBottom]';
        case EVerticalAlign.MIDDLE:
            return 'cssClass[verticalAlignMiddle]';
        case EVerticalAlign.TOP:
            return 'cssClass[verticalAlignTop]';
        case EVerticalAlign.BOTTOM:
            return 'cssClass[verticalAlignBottom]';
    }
};

/**
 * Преобразование значений скрытия колонок в css-классы.
 * @param {EScreenWidth} [size] Массив размеров ширины экрана, при которых скрывать колонку.
 */
export const mapScreenSizeToClassName = (size?: EScreenWidth): string | undefined => {
    switch (size) {
        case EScreenWidth.XS_MAX:
            return 'cssClass[hideXS]';
        case EScreenWidth.SM_MAX:
            return 'cssClass[hideSM]';
        case EScreenWidth.MD_MAX:
            return 'cssClass[hideMD]';
        case EScreenWidth.LG_MAX:
            return 'cssClass[hideLG]';
    }
};

/**
 * Преобразование типа ячейки в css-класс.
 * @param {ECellType} [cellType] Тип ячейки.
 */
export const mapCellTypeToClassName = (cellType: ECellType = ECellType.TEXT): string => cssClass(`${cellType}Type`);
