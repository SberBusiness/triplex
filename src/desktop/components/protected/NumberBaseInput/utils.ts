import {formatAmount} from '@sberbusiness/triplex/desktop/utils/amountUtils';
import {AmountConst} from '@sberbusiness/triplex/desktop/common/consts/AmountConst';

/**
 * Вычисляем позицию каретки.
 *
 * @param {string} prevValueFormatted Прошлое отформатированное значение.
 * @param {string} newValueRaw Новое не отформатированное значение.
 * @param {number} fractionLength Количество знаков в дробной части. При значении 0 подразумевается целочисленное значение.
 * @param {number} currentCaretPosition Позиция каретки в поле ввода.
 */
export const countCaret = (prevValueFormatted: string, newValueRaw: string, fractionLength: number, currentCaretPosition: number) => {
    const spacesInOldValue = (prevValueFormatted.match(/\s/g) || []).length;
    const spacesInNewValue = (formatAmount(newValueRaw, fractionLength).match(/\s/g) || []).length;
    let caretSpaceOffset = spacesInNewValue - spacesInOldValue;

    if (prevValueFormatted.length - newValueRaw.length > 1) {
        // Если было удалено больше одного символа. Делаем reset caretSpaceOffset.
        caretSpaceOffset = 0;
    } else if (newValueRaw.substr(0, 1) === AmountConst.GroupSeparator) {
        // Если впереди пробел (" 786 673") сдвигаем каретку.
        caretSpaceOffset++;
    }

    return currentCaretPosition + caretSpaceOffset;
};
