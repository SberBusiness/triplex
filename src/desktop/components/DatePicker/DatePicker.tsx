import * as React from 'react';
import moment, {Moment} from 'moment';
import {DateInput} from '@sberbusiness/triplex/desktop/components/DatePicker/components/DateInput';
import {inputDateFormat} from '@sberbusiness/triplex/desktop/components/DatePicker/const';
import {Calendar} from '@sberbusiness/triplex/desktop/components/Calendar/Calendar';
import {ICalendarNestedProps} from '@sberbusiness/triplex/desktop/components/Calendar/types';
import {Dropdown, EDropdownAlignment} from '@sberbusiness/triplex/desktop/components/Dropdown/Dropdown';
import {IDateLimitRange} from '@sberbusiness/triplex/desktop/common/types/DateTypes';
import {dateFormatYYYYMMDD, globalLimitRange} from '@sberbusiness/triplex/desktop/common/consts/DateConst';
import {getFormattedDate} from '@sberbusiness/triplex/desktop/utils/dateUtils';
import {EVENT_KEYS} from '@sberbusiness/triplex/desktop/utils/keyboard';
import {classnames} from '@sberbusiness/triplex/common/utils/classnames/classnames';

/** Свойства DatePicker. */
export interface IDatePickerProps
    extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'value' | 'type' | 'onChange' | 'onBlur'>,
        ICalendarNestedProps {
    /** Использование запрещено. */
    children?: never;
    /** Отображаемая по умолчанию дата. */
    defaultViewDate?: string;
    /** Неактивное состояние. */
    disabled?: boolean;
    /** Состояние ошибки. */
    error?: boolean;
    /** Формат для значения. */
    format?: string;
    /** Выравнивание календаря по горизонтали. */
    alignment?: EDropdownAlignment;
    /** Диапазон даты начала и конца периода ограничения. */
    limitRange?: IDateLimitRange;
    /** Отмеченные дни. */
    markedDays?: string[];
    /** Отключенные дни */
    disabledDays?: string[];
    /** Callback для события потери фокуса. */
    onBlur?: () => void;
    /** Callback для события изменения значения. */
    onChange: (date: string) => void;
    /** Символы для заполнения пустых редактируемых позиций в маске (например, строка вида "дд.мм.гггг"). */
    placeholderAsMask?: string;
    /** Обратный порядок выбора даты. */
    reversedPick?: boolean;
    /** Значение даты текстом. */
    value: string;
}

/** Состояния DatePicker. */
interface IDatePickerState {
    /** Value для компонента Calendar. */
    calendarValue: Moment | null;
    /** Фокус в поле открывает popover с выбором даты, а так же позволяет вводить дату вручную. */
    focused: boolean;
    /** Value для компонента DateInput. */
    inputValue: string;
    /** Последняя валидная дата. */
    lastValidValue: string;
    /** Диапазон даты начала и конца периода ограничения. */
    limitRange: IDateLimitRange;
}

/**
 * Проверяет дату на лимиты.
 *
 * @prop {Moment} date Дата для проверки.
 * @prop {IDateLimitRange} limitRange Диапазон даты начала и конца периода ограничения.
 */
const checkOutOfRange = (date: Moment, limitRange: IDateLimitRange): boolean =>
    date.isBefore(limitRange.dateFrom, 'day') || date.isAfter(limitRange.dateTo, 'day');

/** Проверяет дату на disabled. */
const checkDisabled = (inputDate: Moment, disabledDays: string[] | undefined, format: string) => {
    if (!disabledDays) {
        return;
    }

    const inputDateFormatted = inputDate.format(format);
    return !disabledDays.every((disabledDay) => disabledDay !== inputDateFormatted);
};

/** Компонент ввода и выбора даты. */
export class DatePicker extends React.PureComponent<IDatePickerProps, IDatePickerState> {
    public static defaultProps = {
        format: dateFormatYYYYMMDD,
        limitRange: globalLimitRange,
        alignment: EDropdownAlignment.LEFT,
    };

    private readonly targetRef: React.RefObject<HTMLDivElement>;
    private readonly dropdownRef: React.RefObject<HTMLDivElement>;

    constructor(props: IDatePickerProps) {
        super(props);
        const {value, format, limitRange} = props;

        this.state = {
            calendarValue: this.getCalendarValueFromValue(),
            focused: false,
            lastValidValue: moment(value, format, true).isValid() ? value : '',
            inputValue: this.getInputValueFromValue(),
            limitRange: {
                dateFrom: limitRange?.dateFrom || globalLimitRange.dateFrom,
                dateTo: limitRange?.dateTo || globalLimitRange.dateTo,
            },
        };

        this.targetRef = React.createRef();
        this.dropdownRef = React.createRef();
    }

    public componentDidMount(): void {
        document.addEventListener('mousedown', this.handleClickOutside);
        document.addEventListener('touchstart', this.handleClickOutside);
    }

    public componentDidUpdate(prevProps: Readonly<IDatePickerProps>): void {
        const {value, format, limitRange} = this.props;
        const {limitRange: stateLimitRange} = this.state;

        if (value != prevProps.value) {
            this.setLastValidValue(value, format);
            this.setCalendarValue();
            this.setInputValue();
        }

        const dateFrom = limitRange?.dateFrom || globalLimitRange.dateFrom;
        const dateTo = limitRange?.dateTo || globalLimitRange.dateTo;

        if (!stateLimitRange.dateFrom?.isSame(dateFrom) || !stateLimitRange.dateTo?.isSame(dateTo)) {
            this.setState({
                limitRange: {
                    dateFrom,
                    dateTo,
                },
            });
        }
    }

    public componentWillUnmount(): void {
        document.removeEventListener('mousedown', this.handleClickOutside);
        document.removeEventListener('touchstart', this.handleClickOutside);
    }

    public render(): JSX.Element {
        const {
            className,
            error,
            defaultViewDate,
            disabled,
            format,
            limitRange: limitRangeProp,
            markedDays,
            disabledDays,
            onBlur,
            onChange,
            reversedPick,
            value,
            alignment,
            dayHtmlAttributes,
            monthHtmlAttributes,
            yearHtmlAttributes,
            prevButtonProps,
            nextButtonProps,
            changeViewLinkProps,
            placeholderAsMask,
            ...htmlProps
        } = this.props;
        const {calendarValue, focused, inputValue, limitRange} = this.state;

        const classNames = classnames('cssClass[datePicker]', className);

        return (
            <div className={classNames} ref={this.targetRef}>
                <DateInput
                    {...htmlProps}
                    value={inputValue}
                    disabled={disabled}
                    focused={focused}
                    error={error}
                    onFocus={this.handleFocus}
                    onChange={this.handleValueChangeFromInput}
                    onKeyDown={this.handleInputKeyDown}
                    onBlur={this.handleInputBlur}
                    placeholderAsMask={placeholderAsMask}
                />
                <Dropdown opened={focused} targetRef={this.targetRef} forwardedRef={this.dropdownRef} alignment={alignment}>
                    <Calendar
                        pickedDate={calendarValue}
                        limitRange={limitRange}
                        markedDays={markedDays}
                        defaultViewDate={defaultViewDate}
                        disabledDays={disabledDays}
                        onChangeDate={this.handleValueChangeFromCalendar}
                        onBlur={this.handleBlur}
                        reversedPick={reversedPick}
                        dayHtmlAttributes={dayHtmlAttributes}
                        monthHtmlAttributes={monthHtmlAttributes}
                        yearHtmlAttributes={yearHtmlAttributes}
                        prevButtonProps={prevButtonProps}
                        nextButtonProps={nextButtonProps}
                        changeViewLinkProps={changeViewLinkProps}
                    />
                </Dropdown>
            </div>
        );
    }

    /**
     * Обработчик нажатия с клавиатуры.
     * @param {React.KeyboardEvent<HTMLInputElement>} e Событие нажатия клавиши.
     */
    private handleInputKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
        if (EVENT_KEYS.ENTER.includes(e.key)) {
            this.makeValueChangeFromInput();
        }
    };

    private setInputValue = (): void => {
        const {inputValue} = this.state;
        const nextInputValue = this.getInputValueFromValue();

        if (inputValue !== nextInputValue) {
            this.setState({
                inputValue: nextInputValue,
            });
        }
    };

    private getInputValueFromValue = (): string => {
        const {format, value} = this.props;
        let inputValue = value;
        const date = moment(inputValue, format, true);

        if (value && date.isValid()) {
            inputValue = date.format(inputDateFormat);
        }

        return inputValue;
    };

    private setCalendarValue = (): void => {
        const {calendarValue} = this.state;
        const nextCalendarValue = this.getCalendarValueFromValue();

        if (!nextCalendarValue || !calendarValue || !moment(nextCalendarValue).isSame(calendarValue)) {
            this.setState({
                calendarValue: nextCalendarValue,
            });
        }
    };

    private getCalendarValueFromValue = (): Moment | null => {
        const {format, value} = this.props;
        const lastValidValue = this.state ? this.state.lastValidValue : '';
        let calendarValue = null;

        if (value) {
            const date = moment(value, format, true);

            if (date.isValid()) {
                calendarValue = date;
            } else if (lastValidValue) {
                calendarValue = moment(lastValidValue);
            }
        }

        return calendarValue;
    };

    /** Вызов изменения значения по текущему внутреннему состоянию Input. */
    private makeValueChangeFromInput = () => {
        const {onChange, format, disabledDays} = this.props;
        const {inputValue, lastValidValue, limitRange} = this.state;

        if (inputValue === '' && inputValue !== lastValidValue) {
            onChange('');
        }

        const inputDate: Moment = moment(inputValue, inputDateFormat, true);

        const checkSuccess =
            inputDate.isValid() && !checkOutOfRange(inputDate, limitRange) && !checkDisabled(inputDate, disabledDays, format!);
        if (checkSuccess) {
            const newValue = inputDate.format(format);
            if (newValue !== lastValidValue) {
                onChange(newValue);
            }
        } else {
            // возврат состояния к value, которое валидно в этом случае и равно lastValidValue
            this.setCalendarValue();
            this.setInputValue();
        }
    };

    /** Обработчик установки фокуса. */
    private handleFocus = (): void => {
        this.setState({focused: true});
    };

    /** Обработчик потери фокуса. */
    private handleBlur = (): void => {
        this.setState({focused: false});

        // Не обходимо чтобы onBlur был всегда позже onChange (Не правильная последовательность происходит
        // при замене не валидного значения валидным при потере фокуса)
        setTimeout(() => {
            if (this.props.onBlur) {
                this.props.onBlur();
            }
        });
    };

    /**
     * Обработчик потери фокуса поля ввода даты.
     */
    private handleInputBlur = (): void => {
        this.makeValueChangeFromInput();
    };

    /**
     * Обработчик изменения даты из поля для ввода.
     */
    private handleValueChangeFromInput = (event: React.ChangeEvent<HTMLInputElement>): void => {
        const {value} = event.target;
        const inputDate: Moment = moment(event.target.value, inputDateFormat, true);
        if (inputDate.isValid()) {
            this.setState({
                inputValue: value,
                calendarValue: inputDate,
            });
        } else {
            this.setState({
                inputValue: value,
            });
        }
    };

    /**
     * Обработчик изменения даты из календаря.
     * @param {Moment} date Новое значение поля.
     */
    private handleValueChangeFromCalendar = (date: Moment): void => {
        const {onChange, format} = this.props;

        onChange(getFormattedDate(date, format));

        this.handleBlur();
    };

    /** При клике вне компонента скрываем выпадающий список. */
    private handleClickOutside = (event: UIEvent) => {
        const {focused} = this.state;
        const {current: target} = this.targetRef;
        const {current: dropdown} = this.dropdownRef;

        if (focused) {
            if (!target?.contains(event.target as Node) && !dropdown?.contains(event.target as Node)) {
                this.handleBlur();
            }
        }
    };

    /** Заполнение валидного значения даты. */
    private setLastValidValue = (value: string | null, format: string | undefined): void => {
        const {lastValidValue, limitRange} = this.state;
        const {onChange} = this.props;

        const wasChanged = lastValidValue !== value;
        if (!wasChanged) {
            return;
        }

        if (!value) {
            this.setState({lastValidValue: ''});
            return;
        }

        const date = moment(value, format, true);
        const isValid = date.isValid();
        if (!isValid) {
            onChange(lastValidValue);
            return;
        }

        const isOutOfRangeValue = checkOutOfRange(date, limitRange);
        if (isOutOfRangeValue) {
            onChange(lastValidValue);
            return;
        }

        this.setState({lastValidValue: value});
    };
}
