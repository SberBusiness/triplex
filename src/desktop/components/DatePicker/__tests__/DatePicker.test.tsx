import React from 'react';
import moment from 'moment';
import renderer from 'react-test-renderer';
import {getFormattedDate} from '@sberbusiness/triplex/desktop/utils/dateUtils';
import {DatePicker} from '../DatePicker';
import {allure} from '@jest/unit/allure-report';

jest.mock('@sberbusiness/icons/CalendarSrvIcon20', () => ({
    CalendarSrvIcon20: 'svg',
}));

jest.mock('@sberbusiness/icons/CaretleftSrvxIcon24', () => ({
    CaretleftSrvxIcon24: 'svg',
}));

jest.mock('@sberbusiness/icons/CaretrightSrvxIcon24', () => ({
    CaretrightSrvxIcon24: 'svg',
}));

jest.mock('react-text-mask', () => ({
    __esModule: true,
    default: jest.fn(({mask, placeholderChar, ...props}: any) => <input type="text" {...props} />),
    conformToMask: jest.fn((x) => ({conformedValue: x})),
}));

describe('DatePicker', () => {
    beforeEach(() => {
        allure.feature('DatePicker');
    });

    (global as any).document = {
        addEventListener: jest.fn(),
    };

    it('renders correctly', () => {
        const handleChange = jest.fn();
        const value = getFormattedDate(moment('20210101'));
        const tree = renderer.create(<DatePicker value={value} onChange={handleChange} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
