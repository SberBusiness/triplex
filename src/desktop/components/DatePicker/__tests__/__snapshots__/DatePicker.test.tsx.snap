// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders correctly 1`] = `
<div
  className="cssClass[datePicker]"
>
  <div
    className="cssClass[inputBlock]"
  >
    <div
      className="cssClass[maskedInputWrapper]"
    >
      <input
        className="cssClass[input] cssClass[maskedInputPlaceholder] cssClass[dateInput]"
        onChange={[Function]}
        readOnly={true}
        tabIndex={-1}
        value=""
      />
      <input
        className="cssClass[maskedInput] cssClass[dateInput]"
        defaultValue=""
        guide={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        render={[Function]}
        type="text"
        value="01.01.2021"
      />
    </div>
    <span
      className="cssClass[calendarIcon]"
      onMouseDown={[Function]}
    >
      <svg />
    </span>
  </div>
</div>
`;
